### PR TITLE
Update to ensure RFC 2119 compliance

### PR DESCRIPTION
**Pull request by kgryte**
This PR

-   updates the spec to ensure [RFC 2119](https://tools.ietf.org/html/rfc2119) compliance regarding the use of key words to indicate requirement levels.
-   includes various minor typo fixings and formatting fixes.

### CODE REVIEW

File: spec/purpose_and_scope.md [hunk 1]
```diff
@@ -432,3 +432,4 @@ a one-dimensional array.
 The following referenced documents are indispensable for the application of this specification.
 
 -   __IEEE 754-2019: IEEE Standard for Floating-Point Arithmetic.__ Institute of Electrical and Electronic Engineers, New York (2019).
+-   Scott Bradner. 1997. "Keys words for use in RFCs to Indicate Requirement Levels". RFC 2119. doi:[10.17487/rfc2119](https://tools.ietf.org/html/rfc2119).
```
Comment (rgommers) [inline spec/purpose_and_scope.md:435]: minor: typo in first word of title, should be "Key"

**Review by rgommers — APPROVED** [review]
LGTM, thanks @kgryte. One minor typo comment only.

File: spec/purpose_and_scope.md (hunk 1 above)
Comment (kgryte) [reply spec/purpose_and_scope.md]: Fixed. Thanks for catching this!

**Comment by rgommers** [comment]
In it goes!

**Pull request merged**